src/posix/platform/netif.cpp

```diff
@@ -1251,6 +1251,15 @@ static void processNetifAddrEvent(otInstance *aInstance, struct rt_msghdr *rtm)
 
     addr6.sin6_family   = 0;
     netmask.sin6_family = 0;
+    // Skip processing RTM_NEWADDR or RTM_NEWMADDR after interface disabled
+    if (false == otIp6IsEnabled(aInstance))
+    {
+        if ((rtm->rtm_type == RTM_NEWADDR) || (rtm->rtm_type == RTM_NEWMADDR))
+        {
+            error = OT_ERROR_NONE;
+            goto exit;
+        }
```

Comment:
Conceptually even if netif is down we can still add/remove addresses on netif. I know OT APIs (like `otIp6AddUnicastAddress()`) support this.

Reply:
thanks @abtink can you please help share why add address should be supported after netif down? 

Most likely thread stop is also called on netif down, which would have cleaned up all the addresses and multicast group subscriptions right? If the RTM_NEWMADDR event is handled and adds back the multicast address (after thread stop), this would never get cleaned up?

During the next netif up and thread start, while adding the multicast address back (IPV6_JOIN_GROUP), it can lead to an error?

Reply:
I am not super familiar with the `platform/netif.cpp` code itself, so maybe we need a fix here to handle the multicast case? Or what you propose is right for this?

> can you please help share why add address should be supported after netif down?

From what I recall, there were some use cases where users wanted to add a (fixed or known) address before bringing the interface up.  I know OT APIs allow this use:
```
> ifconfig
down
Done

> ipaddr add fd00::beef:cafe
Done
> ipaddr add fd00::1111
Done

> ipaddr
fd00:0:0:0:0:0:0:1111
fd00:0:0:0:0:0:beef:cafe
Done

> ipaddr del fd00::1111
Done

> ipaddr
fd00:0:0:0:0:0:beef:cafe
Done

> ipmaddr add ff0a::1
Done

> ifconfig up
Done

> ipaddr
fe80:0:0:0:e03c:54e1:d6bf:a18b
fd00:0:0:0:0:0:beef:cafe
Done

> ifconfig 
up
Done

> thread start
Done

> state
leader
Done

> ipaddr
fdb7:6ea9:e4e5:9fe8:0:ff:fe00:fc00
fdb7:6ea9:e4e5:9fe8:0:ff:fe00:f800
fdb7:6ea9:e4e5:9fe8:fa14:b74d:de66:5ff4
fe80:0:0:0:e03c:54e1:d6bf:a18b
fd00:0:0:0:0:0:beef:cafe
Done

> ipmaddr
ff33:40:fdfb:7b4c:9b6e:3cfb:0:1
ff32:40:fdfb:7b4c:9b6e:3cfb:0:1
ff0a:0:0:0:0:0:0:1
ff02:0:0:0:0:0:0:2
ff03:0:0:0:0:0:0:2
ff02:0:0:0:0:0:0:1
ff03:0:0:0:0:0:0:1
ff03:0:0:0:0:0:0:fc
```

 and if I am not mistaken linux `ip -6 ` command also allows this (i.e., we can add/remove address to  a netif that is down).

Reply:
@sarveshkumarv3 , any further thoughts?

Reply:
@jwhui @abtink what is the reasoning to allow add address when the netif is down? is there any scenario where this would be exercised?

Reply:
I mentioned above, I recall there were some use-cases (within Google) were pre-known/fixed (app specified) addresses were added by app layer code on device on start (which could happen while netif was still down). Not sure if they are still applicable or in-use, but  OT stack allows this model of use, for sure. There may be other folks that may be using/relying on similar behavior as well. 

I understand there may need to be some changes in how posix netlink is woking today  but the question for me is if disallowing address add/remove when netif is down is the right approach? (as this can potentially cause compatibility issues?)

Reply:
thanks for the clarification, it was not clear to me why assigning an address when netif down would help. But sure, I am ok if we close this PR without merging, if handling address add events after interface down should be supported.